src/content/PagePartials/index.ts

```diff
@@ -0,0 +1,57 @@
+import { defineCollection, z } from 'astro:content';
+import {
+  PagePartialCollectionEntry as query,
+  type PagePartialCollectionEntryQuery,
+  type SiteLocale
+} from '@lib/datocms/types.ts';
+import { datocmsCollection, datocmsRequest } from '@lib/datocms';
+
+export type PagePartialCollectionEntry = PagePartialCollectionEntryQuery['entry'] & {
+  recordId: string,
+  id: string,
+  locale: SiteLocale,
+};
+
+const name = 'PagePartials' as const;
+
+const loadEntry = async ({ id, locale }: { id: string, locale: SiteLocale }) => {
+  const { entry } = await datocmsRequest<PagePartialCollectionEntryQuery>({ query, variables: { id, locale } });
+  return {
+    ...entry,
+    recordId: entry.id,
+    id: `${locale}/${entry.id}`,
+    locale,
+  } satisfies PagePartialCollectionEntry;
+};
+
+const loadCollection = async () => {
+  const items = (await datocmsCollection<{ 
+    id: string, 
+    _allBlocksLocales: { locale: SiteLocale }[] 
+  }>({
+    collection: name,
+    fragment: /* graphql */`
+      id
+      _allBlocksLocales {
+        locale
+      }
+    `,
+  })).flatMap(({ id, _allBlocksLocales }) => _allBlocksLocales.map(({ locale }) => ({ id, locale })));
```

Comment:
Maybe add an explanation inline why this is needed? Or on top of the function add add some JSDoc with an example of the return value? I feel like the abstractions and mappings that we're introducing in this file might be hard to understand for the next developer (I'm already struggling a bit and we've discussed this before).